master/buildbot/db/steps.py

```diff
@@ -151,10 +151,15 @@ def thd(conn):
             if _racehook is not None:
                 _racehook()
             urls = json.loads(row.urls_json)
-            urls.append(dict(name=name, url=url))
 
-            q = tbl.update(whereclause=wc)
-            conn.execute(q, urls_json=json.dumps(urls))
+            url_exists = False
```

Comment:
this code is not very pythonic.

I would compare the whole object anyway, and this helps to simplify the code 

```
url = dict(name=name, url=url)
if url not in urls:
    [...]
```